### PR TITLE
KT-13146 J2K recursion while converting self-referenced anonymous functions

### PR DESCRIPTION
**Pull request by semoro**
https://youtrack.jetbrains.com/issue/KT-13146

This fix is only small workaround of a big problem

### CODE REVIEW

File: j2k/src/org/jetbrains/kotlin/j2k/ExpressionConverter.kt [hunk 1]
```diff
@@ -538,6 +538,13 @@ class DefaultExpressionConverter : JavaElementVisitor(), ExpressionConverter {
         val qualifier = expression.qualifierExpression
 
         var identifier = Identifier.withNoPrototype(referenceName, isNullable)
+
+
+        if (expression.getContainingClass()?.getParentOfType<PsiField>(false).let { it != null && it == expression.qualifierExpression?.reference?.resolve() }) {
```
Comment (yole) [inline j2k/src/org/jetbrains/kotlin/j2k/ExpressionConverter.kt]: Why are you checking for `PsiField` specifically? Won't the same problem happen if the anonymous class is used as a local variable initializer?

Comment (yole) [inline j2k/src/org/jetbrains/kotlin/j2k/ExpressionConverter.kt]: `x.let { it != null && ... }` is much better expressed as `x?.let { ... }`

Comment (semoro) [inline j2k/src/org/jetbrains/kotlin/j2k/ExpressionConverter.kt]: It is used in if check
Which better
`if(x.let{ it != null && ........................long_check.....................................})`
or
`if(x?.let{  ........................long_check..................................... } ?: false)`

Comment (semoro) [inline j2k/src/org/jetbrains/kotlin/j2k/ExpressionConverter.kt]: I should use `PsiVariable` to cover fields and local variables?

**Comment by yole** [comment]
Merged into master and 1.0.5. Thanks for your contribution!

**Pull request merged**